File: first_non_repeating_character/first_non_repeating_character.go
package main

import (
    "fmt"
)

func firstNonRepeatingCharacter(s string) int {
  characters := make(map[rune]bool)
  for _, c := range s {
    if _, exists := characters[c]; exists {
      characters[c] = false
    } else {
      characters[c] = true
    }
  }
  for idx, c := range s {
    if characters[c] {
      return idx
    }
  }
  return -1
}

func main() {
  fmt.Println(firstNonRepeatingCharacter("abcdcaf"))
}

File: first_non_repeating_character/first_non_repeating_character.py
def first_non_repeating_character(string):
  characters = {}
  for character in string:
    if character in characters:
      characters[character] = False
    else:
      characters[character] = True
  for idx, character in enumerate(string):
    if character in characters and characters[character] == True:
      return idx
  return -1

print(first_non_repeating_character("abcdcaf"))